### PR TITLE
Associations and queries

### PR DESCRIPTION
**Pull request by SimbaWANGU**
In this Pull Request, the following list of requirements were completed:
1. Created models
2. Used models to insert data using `rails c` command
3. Created the following custom methods:
    - A method that returns the 3 most recent posts for a given user.
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
    - A method that updates the comments counter for a post.
    - A method that updates the likes counter for a post.

### CODE REVIEW

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @SimbaWANGU, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 - Good Readme :heavy_check_mark: 
 - No linter error :heavy_check_mark: 
 - Good PR title & summary :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/author.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :comments
+  has_many :posts, through: :likes
+
+  def most_recent_posts
+    Post.limit(3).order(created_at: :desc).where(author: self)
+  end
+
+  def update_post_counter
+    update(posts_counter: Post.where(author: self).count)
+  end
```
[the remark this answers is not in the document]
Comment (SimbaWANGU) [reply app/models/author.rb]: Thank you very much for this illustration @mmsesay 👍

**Review by elmar8287 — APPROVED** [review]
Hi, @SimbaWANGU.

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 

### Highlights

- You have implemented all suggestions from the previous reviewer 👍 

### Optional suggestions

- Please, consider fixing the broken link to the MIT.md file 👍 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SimbaWANGU** [comment]
@elmar8287 thank you for the review 👍

**Pull request merged**